pkgs/development/tools/fastddsgen/default.nix

```
@@ -33,12 +44,12 @@ stdenv.mkDerivation {
 
     gradle install --install_path=$out
 
-    # Override the default start script to use absolute java path
-    cat  <<EOF >$out/bin/fastddsgen
-    #!${runtimeShell}
-    exec ${openjdk17}/bin/java -jar "$out/share/fastddsgen/java/fastddsgen.jar" "\$@"
-    EOF
-    chmod a+x "$out/bin/fastddsgen"
+    # Override the default start script to use absolute java path.
+    # Make the unwrapped "cpp" available in the path, since the wrapped "cpp"
+    # passes additional flags and produces output incompatible with fastddsgen.
+    makeWrapper ${openjdk17}/bin/java $out/bin/fastddsgen \
+      --add-flags "-jar $out/share/fastddsgen/java/fastddsgen.jar" \
+      --prefix PATH ${lib.makeBinPath [ stdenv.cc.cc ]}
```

```suggestion
      --prefix PATH : ${lib.makeBinPath [ stdenv.cc.cc ]}
```

You should add a separator (`:`). Without this, `PATH` is not modified at all.